### PR TITLE
[Shop][Checkout] Allow change payment method for new or pending payment

### PR DESCRIPTION
**Pull request by tuka217**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Related tickets |-
| License         | MIT

### CODE REVIEW

File: src/Sylius/Bundle/ShopBundle/Resources/views/Order/payment.html.twig [hunk 1]
```diff
@@ -1,22 +1,20 @@
 {{ form_start(form, {'action': path('sylius_shop_order_payment_method_update', {'orderId': order.id}), 'attr': {'novalidate': 'novalidate'}}) }}
 <input type="hidden" name="_method" value="PUT" />
-<div class="ui segment">
-    <div class="sub header">{{ 'sylius.ui.change_payment_method'|trans }}</div>
-    <div class="ui fluid stackable items">
-        {% set payment_form = form.payments[order.payments|length -1] %}
-        {{ form_errors(payment_form.method) }}
+<div class="sub header">{{ 'sylius.ui.change_payment_method'|trans }}</div>
+{% set payment_form = form.payments[order.payments|length -1] %}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Order/payment.html.twig]: `{% set payment_form = form.payments|last %}`?

File: src/Sylius/Bundle/ShopBundle/Resources/views/Order/payment.html.twig [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
         {% for payment_method_choice in payment_form.method %}
-            <div class="item">
-                <div class="field">
-                    <div class="ui radio checkbox">
-                        {{ form_widget(payment_method_choice) }}
-                    </div>
+            <div class="field">
+                <div class="ui radio checkbox">
+                    {{ form_widget(payment_method_choice) }}
+                    <label>{{ payment_method_choice.vars.label}}</label>
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Order/payment.html.twig]: Missing space at the end of `}}`

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig [hunk 3]
```diff
@@ -1,38 +1,53 @@
...
+                            </div>
+                        {% elseif payment.state is same as('pending') %}
+                            <i class="circular rocket icon"></i>
+                                <div class="ui centered grid">
+                                    <div class="left aligned column">
+                                    {{ render(url('sylius_shop_order_payment_method_update', {'orderId': order.id} )) }}
+                                    </div>
+                                </div>
+                            <div class="ui hidden divider"></div>
+                        {% else %}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: Making it an `elseif` will make life easier for those that add custom states.

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig [hunk 4]
```diff
@@ -1,38 +1,53 @@
...
+                                            {{ 'sylius.ui.pay'|trans }}
+                                        </a>
+                                    </div>
+                                </div>
+                            </div>
+                        {% elseif payment.state is same as('pending') %}
+                            <i class="circular rocket icon"></i>
+                                <div class="ui centered grid">
+                                    <div class="left aligned column">
+                                    {{ render(url('sylius_shop_order_payment_method_update', {'orderId': order.id} )) }}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: `{{ render(url('sylius_shop_order_payment_method_update', {'orderId': order.id})) }}`

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig [hunk 5]
```diff
@@ -1,38 +1,53 @@
...
-                        </a>
-                    </div>
-                {% endif %}
-                {% if lastPayment.method.instructions is not null %}
-                    <div id="sylius-payment-method-instructions" class="sub header">
-                        {{ lastPayment.method.instructions }}
-                    </div>
-                {% endif %}
+                <div class="ui segment">
+                    {% for payment in order.payments if payment.state is not same as('cancelled')%}
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: Missing space before `%}`

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig [hunk 6]
```diff
@@ -1,38 +1,53 @@
...
+                                        {{ render(url('sylius_shop_order_payment_method_update', {'orderId': order.id} )) }}
+                                    </div>
+                                    <div class="ui vertical divider">
+                                        or
+                                    </div>
+                                    <div class="column">
+                                        <div class="sub header">{{ 'sylius.ui.payment_ready_to_pay'|trans }} </div>
+                                        <div class="ui hidden divider"></div>
+                                        <div class="ui hidden divider"></div>
+                                        <div class="ui hidden divider"></div>
```
Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: Why do we need three of them?

File: src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig (hunk 3 above)
Comment (tuka217) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: But what do you pass as condition?

Comment (pamil) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Checkout/thankYou.html.twig]: The state that is required to show thank you page (I guess payment should be completed by then).

**Pull request closed without merging**